### PR TITLE
Page navigation events

### PR DESCRIPTION
**Pull request by vakrilov**
Related to #176

### CODE REVIEW

**Comment by enchev** [comment]
:+1:

**Comment by ErjanGavalji** [comment]
:+1:

File: ui/page/page-common.ts [hunk 1]
```diff
@@ -96,23 +99,37 @@ export class Page extends contentView.ContentView implements dts.Page, view.AddA
 
     public onNavigatingTo(context: any) {
```
Comment (hshristov) [inline ui/page/page-common.ts]: Do we need context in onNavigatingTo?

File: ui/page/page.d.ts [hunk 2]
```diff
@@ -14,11 +14,11 @@ declare module "ui/page" {
     //@endprivate
 
     /**
-     * Defines the data for the Page.navigatedTo event.
+     * Defines the data for the page navivation events.
```
Comment (hshristov) [inline ui/page/page.d.ts]: Wrong word - "navivation"

File: ui/page/page.d.ts [hunk 3]
```diff
@@ -95,7 +115,7 @@ declare module "ui/page" {
          * A method called after navigated to the page.
          * @param context - The data passed to the page through the NavigationEntry.context property.
```
Comment (hshristov) [inline ui/page/page.d.ts]: @param context should be removed from API reference help.

**Pull request merged**